Xwt.Gtk/Xwt.GtkBackend/RichTextViewBackend.cs

```diff
@@ -382,36 +389,219 @@ public void EmitHorizontalRuler ()
+				get {
+					if (defaultCursor == null)
+						defaultCursor = new Gdk.Cursor (Gdk.CursorType.Xterm);
+					return defaultCursor;
+				}
+			}
+
+			Gdk.Cursor HandCursor {
+				get {
+					if (handCursor == null)
+						handCursor = new Gdk.Cursor (Gdk.CursorType.Hand1);
+					return handCursor;
+				}
+			}
+
+			protected override bool OnMotionNotifyEvent (Gdk.EventMotion evnt)
+			{
+				Link link = GetLinkAtPos (evnt.X, evnt.Y);
+				if (link != null) {
+					if (activeLink == null) {
+						TooltipText = link.Title;
+
+						GetWindow (Gtk.TextWindowType.Text).Cursor = HandCursor;
+					} else if (activeLink.Title != link.Title) {
+						TooltipText = link.Title;
+					}
+					activeLink = link;
+				} else if (activeLink != null) {
+					activeLink = null;
+					TooltipText = null;
+					SetDefaultCursor ();
+				}
+
+				if (selectable)
+					return base.OnMotionNotifyEvent (evnt);
+				return true;
+			}
+
+			protected override bool OnButtonPressEvent (Gdk.EventButton evnt)
+			{
+				if (selectable)
+					return base.OnButtonPressEvent (evnt);
+				return true;
+			}
+
+			protected override bool OnButtonReleaseEvent (Gdk.EventButton evnt)
+			{
+				if (NavigateToUrl != null && (PointerButton)evnt.Button == PointerButton.Left) {
+					Link link = activeLink ?? GetLinkAtPos (evnt.X, evnt.Y);
+
+					if (link?.Href != null)
+						NavigateToUrl?.Invoke (this, new NavigateToUrlEventArgs (link.Href));
+				}
+				if (selectable)
+					return base.OnButtonReleaseEvent (evnt);
+				return true;
+			}
+
+			Link GetLinkAtPos (double mousex, double mousey)
+			{
+				int x, y;
+				WindowToBufferCoords (Gtk.TextWindowType.Text, (int)mousex, (int)mousey, out x, out y);
+				var iter = GetIterAtLocation (x, y);
+				foreach (var l in Buffer.Links) {
+					if (iter.HasTag (l.Key)) {
+						return l.Value;
 					}
-					while (attrIter.Next ());
 				}
+				return null;
+			}
+
+			protected override void OnStateChanged (Gtk.StateType previous_state)
+			{
+				base.OnStateChanged (previous_state);
+				SetDefaultCursor ();
+				UpdateBackground ();
+			}
+
+			protected override void OnGrabNotify (bool was_grabbed)
+			{
+				base.OnGrabNotify (was_grabbed);
+				if (!was_grabbed)
+					SetDefaultCursor ();
+			}
+
+			protected override void OnRealized ()
+			{
+				base.OnRealized ();
+				SetDefaultCursor ();
+				UpdateBackground ();
+				UpdateLinkColor ();
+			}
+
+			protected override void OnStyleSet (Gtk.Style previous_style)
+			{
+				base.OnStyleSet (previous_style);
+				UpdateLinkColor ();
+			}
+
+			void SetDefaultCursor ()
+			{
+				if (!IsRealized)
+					return;
+				Gdk.Cursor cursor = Sensitive && Selectable ? DefaultCursor : null;
+				GetWindow (Gtk.TextWindowType.Text).Cursor = cursor;
+
+			}
+
+			void UpdateLinkColor ()
+			{
+				if (!IsRealized)
+					return;
+				var color = (Gdk.Color) StyleGetProperty ("link-color");
+				if (color.Equals (Gdk.Color.Zero))
+					color = Colors.Blue.ToGtkValue ();
```

Comment:
Since we use `Colors.Blue.ToGtkValue ()` in 3 places, maybe make it an instance variable?